src/Rules/Abstracts/ConditionalRule.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace StellarWP\Validation\Rules\Abstracts;
+
+use StellarWP\FieldConditions\ComplexConditionSet;
+use StellarWP\FieldConditions\Contracts\Condition;
+use StellarWP\FieldConditions\Contracts\ConditionSet;
+use StellarWP\FieldConditions\SimpleConditionSet;
+use StellarWP\Validation\Config;
+use StellarWP\Validation\Contracts\ValidatesOnFrontEnd;
+use StellarWP\Validation\Contracts\ValidationRule;
+
+abstract class ConditionalRule implements ValidationRule, ValidatesOnFrontEnd
+{
+    /**
+     * @var ConditionSet
+     */
+    protected $conditions;
+
+    /**
+     * @param ConditionSet|Condition[] $conditions
+     */
+    public function __construct($conditions)
+    {
+        if ($conditions instanceof ConditionSet) {
+            $this->conditions = $conditions;
+        } else {
+            $this->conditions = new ComplexConditionSet(...$conditions);
+        }
```

Comment:
I wonder if we should allow allow conditions to be overwritten directly on an existing Rule by moving this to a method and calling it here on construct.

Reply:
Unless you have a clear and immediate case in mind, I'm inclined to keep this simple until a use case arrises.